net/Socket.hpp

```diff
@@ -1257,8 +1257,21 @@ class StreamSocket : public Socket,
                         last_errno = errno; // Save only on error.
 
                     if (len < 0 && last_errno != EAGAIN && last_errno != EWOULDBLOCK)
-                        LOG_SYS_ERRNO(last_errno, "Read failed, have " << _inBuffer.size()
-                                                                       << " buffered bytes");
+                    {
+                        if (closed && last_errno == ECONNRESET)
```

Comment:
I'm really not convinced by the complexity of the extra parameter - and there is of course a race between the poll returning and the socket closing - so this will not be a reliable catch here.

We will be able to see the socket closed with a HUP in other logging - to diagnose why.

I would be inclined to just LOGA_INF() or somesuch for all ECONNRESETs in this case - which should simplify the patch a lot.

Is that do-able ?

Reply:
Yes, even better, since we can have the same effect by checking the buffer.

That is, if there is nothing in the buffer, then the connection was idle and the peer recycled the socket.

I also reduced to trace logging to match the others in the function.